src/main/java/com/slavalapin/ifmo/web/servlets/CalculatorInternal.java

```diff
@@ -0,0 +1,124 @@
+package com.slavalapin.ifmo.web.servlets;
+
+import java.util.*;
+import java.util.regex.Pattern;
+
+
+public class CalculatorInternal {
+
+    public static int doMagic(String expressionOriginal, Map<String, String> variablesMap)
+            throws MissingResourceException
+    {
+        ArrayDeque<String> expressionReady = mapVariables(toPolishNotation(expressionOriginal), variablesMap);
+        return compute(expressionReady);
+    }
+
+    private static int compute(ArrayDeque<String> expression) {
+        Stack<Integer> calculationStack = new Stack<>();
+        while (!expression.isEmpty()) {
+            String step = expression.poll();
+            if(Pattern.matches("-?[0-9]+", step))
+                calculationStack.push(Integer.parseInt(step));
+            else {
+                char symbol = step.charAt(0);
+                int rightOperand = calculationStack.pop();
+                int leftOperand = calculationStack.pop();
+                int resultIntermediate;
+                switch (symbol) {
+                    case '+':
+                        resultIntermediate = rightOperand + leftOperand;
+                        break;
+                    case '-':
+                        resultIntermediate = leftOperand - rightOperand;
+                        break;
+                    case '*':
+                        resultIntermediate = rightOperand * leftOperand;
+                        break;
+                    case '/':
+                        resultIntermediate = leftOperand / rightOperand;
+                        break;
+                    default:
+                        throw new java.lang.IllegalStateException("Unexpected value: " + step);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'java.lang.IllegalStateException' due to existing implicit import 'java.lang.*'](https://app.codacy.com/manual/thejerome/IFMO_Servlets_20190915/pullRequest?prid=5345482)